### PR TITLE
status: show a progress bar while gathering info

### LINKED ISSUE #1762: perf: remote: status: don't pull full list of files from remote
Currently, we retrieve the full list of files from remote and then compare it with local cache, which gets really slow if user has lots of files on the remote already. So instead of doing this https://github.com/iterative/dvc/blob/0.32.1/dvc/remote/local.py#L591 we should compare it with local cache more carefully and efficiently. 

https://discordapp.com/channels/485586884165107732/485596304961962003/558288115249184775

### PR DESCRIPTION
**Pull request by mroutis**
Fix #1762
Fix https://github.com/iterative/dvc/issues/1825

* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

### CODE REVIEW

**Comment by efiop** [comment]
@mroutis Travis tests are failing.

File: dvc/remote/local/__init__.py [hunk 1]
```diff
@@ -360,37 +365,23 @@ def status(
...
-        progress.update_target(title, 0, 100)
-
-        progress.update_target(title, 10, 100)
-
         ret = self._group(checksum_infos, show_checksums=show_checksums)
         md5s = list(ret)
 
-        progress.update_target(title, 30, 100)
-
+        logger.info("Collecting information for local cache...")
```
Comment (efiop) [inline dvc/remote/local/__init__.py]: How about we use this message (or maybe something like `Collecting info for s3://mybucket/path`) in the progress bars themselves?

File: dvc/remote/base.py [hunk 2]
```diff
@@ -532,13 +532,20 @@ def cache_exists(self, checksums):
         Returns:
             A list with checksums that were found in the remote
         """
+        progress_callback = ProgressCallback(len(checksums))
+
+        def exists_with_progress(chunks):
+            return self.exists(chunks, callback=progress_callback)
```
Comment (efiop) [inline dvc/remote/base.py]: I don't see this progress callback being used anywhere.

File: dvc/remote/local/__init__.py (hunk 1 above)
Comment (mroutis) [reply dvc/remote/local/__init__.py]: @efiop , instead of showing the checksums?

File: dvc/progress.py [hunk 3]
```diff
@@ -151,4 +151,14 @@ def finish(self, name):
         progress.finish_target(name)
 
 
+def progress_wrapper(iterable):
```
Comment (efiop) [inline dvc/progress.py]: Isn't `Progress.__call__` the same thing? E.g. https://github.com/iterative/dvc/blob/master/dvc/remote/base.py#L148

Comment (mroutis) [reply dvc/progress.py]: :eyes: ! It is, @efiop , good catch

**Pull request merged**